src/accelerate/utils/memory.py

```diff
@@ -159,3 +162,16 @@ def decorator(*args, **kwargs):
                     raise
 
     return decorator
+
+
+def get_xpu_available_memory(device_index: int):
+    if is_ipex_available():
+        ipex_version = importlib.metadata.version("intel_extension_for_pytorch")
+        if compare_versions(ipex_version, ">=", "2.5"):
+            from intel_extension_for_pytorch.xpu import mem_get_info
+
+            return mem_get_info(device_index)[0]
+        warnings.warn(
```

Comment:
Just to be sure: When ipex is _not_ available, `torch.xpu.max_memory_allocated(device_index)` will return the correct result? I'm asking since the warning is only given if ipex is available.

Reply:
you are right! `torch.xpu.max_memory_allocated(device_index)` doesn't return the correct result. So I updated the warning. Thx!!